### ISSUE TITLE
Can't trigger a backup from HTTP GET without a user logged in

### ISSUE THREAD

**Issue by jimtremblay** [gitbucket/gitbucket-h2-backup-plugin#8]
If there is no user logged in at the time I call `wget http://localhost:8080/database/backup` to trigger a backup, I receive the sign-in html page and no backup is made. It prevents full backup from script if there is no user logged in. I'm using GitBucket 3.8 and h2 plugin 1.0

How to avoid this?

Thanks
Jim

**Comment by McFoggy** [comment]
I didn't had the opportunity to test 3.8 but the plugin doesn't require authentication as shown in source [code](https://github.com/gitbucket-plugins/gitbucket-h2-backup-plugin/blob/master/src/main/scala/fr/brouillard/gitbucket/h2/controller/H2BackupController.scala#L43) ; moreover such a security is the issue #1 that expected gitbucket core to offer http basic authentication.

I'll have to check if gitbucket core code didn't introduced some security checks.

**Comment by McFoggy** [comment]
@jimtremblay I just tested with gitbucket 3.8 & gitbucket-h2-backup-plugin 1.0.
I do not reproduce your problem.
Can you provide more context please.

Here is what I did with a fresh new installation.
I launched gitbucket using this syntax:

```
SET GITBUCKET_HOME=D:\dev\projects\oss\gitbucket\releases\3.8\home
java -jar D:\dev\projects\oss\gitbucket\releases\3.8\gitbucket.war
```

and here are the results I got when calling the backup

```
D:\dev\projects\oss\gitbucket\releases\3.8>ls
gitbucket.war  home           launch.bat

D:\dev\projects\oss\gitbucket\releases\3.8>ls home
data.mv.db  plugins     tmp         version

D:\dev\projects\oss\gitbucket\releases\3.8>ls home\plugins
gitbucket-h2-backup-plugin.jar

D:\dev\projects\oss\gitbucket\releases\3.8>wget http://localhost:8080/database/backup
--09:34:09--  http://localhost:8080/database/backup
           => `backup'
Resolving localhost... done.
Connecting to localhost[127.0.0.1]:8080... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4 [text/html]

100%[====================================>] 4              3.91K/s    ETA 00:00

09:34:09 (3.91 KB/s) - `backup' saved [4/4]


D:\dev\projects\oss\gitbucket\releases\3.8>more backup
done

D:\dev\projects\oss\gitbucket\releases\3.8>ls home
data.mv.db                     tmp
gitbucket-database-backup.zip  version
plugins

D:\dev\projects\oss\gitbucket\releases\3.8>
```

**Comment by jimtremblay** [comment]
I'm on Ubuntu Server 14.04

I have created a directory /usr/share/gitbucket and put gitbucket.war in it.
This is how gitbucket is started: `GITBUCKET_HOME="/var/lib/gitbucket" java -Dmail.smtp.starttls.enable=true -jar /usr/share/gitbucket/gitbucket.war --port=8080 --host=0.0.0.0`

I don't log any user from web browser and I'm calling directly `wget http://localhost:8080/database/backup` to do the backup, here what I have:

```
--2015-11-13 06:44:16--  http://localhost:8080/database/backup
Resolving localhost (localhost)... ::1, 127.0.0.1
Connecting to localhost (localhost)|::1|:8080... connected.
HTTP request sent, awaiting response... 302 Found
Location: http://localhost:8080/signin;jsessionid=mv933zsynhmz65dayx45qzyq?redirect=%2Fdatabase%2Fbackup [following]
--2015-11-13 06:44:16--  http://localhost:8080/signin;jsessionid=mv933zsynhmz65dayx45qzyq?redirect=%2Fdatabase%2Fbackup
Reusing existing connection to [localhost]:8080.
HTTP request sent, awaiting response... 200 OK
Length: 3933 (3.8K) [text/html]
Saving to: 'backup'

100%[======================================>] 3,933       --.-K/s   in 0s      

2015-11-13 06:44:16 (78.3 MB/s) - 'backup' saved [3933/3933]
```

`cat backup` give:

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title>Sign in</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <link rel="icon" href="/assets/common/images/gitbucket.png" type="image/vnd.microsoft.icon" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link href="/assets/vendors/bootstrap/css/bootstrap.css" rel="stylesheet">
    <link href="/assets/vendors/bootstrap/css/bootstrap-responsive.css" rel="stylesheet">
    <link href="/assets/vendors/octicons/octicons.css" rel="stylesheet">
    <link href="/assets/vendors/datepicker/css/datepicker.css" rel="stylesheet">
    <link href="/assets/vendors/colorpicker/css/bootstrap-colorpicker.css" rel="stylesheet">
    <link href="/assets/vendors/google-code-prettify/prettify.css" type="text/css" rel="stylesheet"/>
    <link href="/assets/vendors/facebox/facebox.css" rel="stylesheet"/>
    <link href="/assets/common/css/gitbucket.css" rel="stylesheet">
    <script src="/assets/vendors/jquery/jquery-1.11.1.js"></script>
    <script src="/assets/vendors/dropzone/dropzone.js"></script>
    <script src="/assets/common/js/validation.js"></script>
    <script src="/assets/common/js/gitbucket.js"></script>
    <script src="/assets/vendors/bootstrap/js/bootstrap.js"></script>
    <script src="/assets/vendors/datepicker/js/bootstrap-datepicker.js"></script>
    <script src="/assets/vendors/colorpicker/js/bootstrap-colorpicker.js"></script>
    <script src="/assets/vendors/google-code-prettify/prettify.js"></script>
    <script src="/assets/vendors/zclip/ZeroClipboard.min.js"></script>
    <script src="/assets/vendors/elastic/jquery.elastic.source.js"></script>
    <script src="/assets/vendors/facebox/facebox.js"></script>
    <script src="/assets/vendors/jquery-hotkeys/jquery.hotkeys.js"></script>

  </head>
  <body>
    <form id="search" action="/search" method="POST">
      <div class="navbar">
        <div class="navbar-inner">
          <div class="container" style="width: 980px;">
            <button type="button" class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a class="brand" href="/">
              <img src="/assets/common/images/gitbucket.png" style="width: 24px; height: 24px;"/>GitBucket

                <span class="header-version">3.8</span>


            </a>


            <div class="nav-collapse collapse pull-right header-menu">

                <a href="/signin?redirect=%2Fsignin%3Bjsessionid%3Dmv933zsynhmz65dayx45qzyq" class="btn" id="signin">Sign in</a>

            </div><!--/.nav-collapse -->
          </div>
        </div>
      </div>
    </form>

  <div class="signin-form">


<div class="box-header">

  <span class="strong">Sign in</span>
</div>
<div class="box-content-bottom">
  <form action="/signin" method="POST" validate="true">
    <label for="userName">Username:</label>
    <span id="error-userName" class="error"></span>
    <input type="text" name="userName" id="userName" style="width: 95%" autofocus/>
    <label for="password">Password:</label>
    <span id="error-password" class="error"></span>
    <input type="password" name="password" id="password" style="width: 95%"/>
    <div>
      <input type="submit" class="btn btn-success" value="Sign in"/>
    </div>
  </form>
</div>

  </div>

    <script>
      $(function(){
        $('#search').submit(function(){
          return $.trim($(this).find('input[name=query]').val()) != '';
        });
      });
    </script>

      <script>

$('a.global-header-menu:last').after(
  $('<a href="/gist" class="global-header-menu">Gist</a>')
);

      </script>

  </body>
</html>
```

**Comment by McFoggy** [comment]
@jimtremblay I have no ubuntu to test it, but as I have some docker images I quickly created a docker image for it (debian based, not really ubuntu) ; docker file [here](https://gist.github.com/McFoggy/2c5180a219d5834e7bc7)
- `docker build -t docker-gitbucket .`
- `docker run -p 8080:8080 docker-gitbucket /usr/bin/java -Dmail.smtp.starttls.enable=true -jar gitbucket.war --port=8080 --host=0.0.0.0`

I can successfully do a wget on `http://my-docker-box-ip:8080/database/backup`

**Comment by jimtremblay** [comment]
I have just try your step and it works for me too.

I think I have just found the cause: Anonymous access is disabled. From my instance on Ubuntu, if I enable anonymous access, everything went well.

Maybe a secret key can be passed by URL to allow backup or access from localhost is always accepted even if anonymous access is disabled.

Thanks
Jim

**Comment by McFoggy** [comment]
without gitbucket/gitbucket#430 I do not see what I can do in the plugin.

currently the `database/backup` endpoint is unsecured especially because there is no possibility to secure it in a `script` friendly way (token, basic auth, ...)

**Comment by McFoggy** [comment]
closed as duplicate of #1
